pkg/logservice/rsm.go

```diff
@@ -202,14 +202,20 @@ func (s *stateMachine) Lookup(query interface{}) (interface{}, error) {
 
 func (s *stateMachine) SaveSnapshot(w io.Writer,
 	_ sm.ISnapshotFileCollection, _ <-chan struct{}) error {
-	// FIXME: use gogoproto to marshal the state, need to figure out how to
-	// marshal to a io.Writer
-	enc := gob.NewEncoder(w)
-	return enc.Encode(s.state)
+	data := make([]byte, s.state.Size())
```

Comment:
snapshot creation is currently designed to be very frequent, and mpool should be used here to avoid frequent make memory

Reply:
This PR mainly replace encoding/gob, I suggest that leaving further optimization to related co-workers and
optimizing when this would be a bottleneck.